src/main/java/no/nav/fo/veilarbdialog/db/dao/DataVarehusDAO.java

```diff
@@ -1,78 +1,47 @@
 
-import no.nav.common.auth.SubjectHandler;
+import lombok.RequiredArgsConstructor;
 import no.nav.fo.veilarbdialog.domain.DatavarehusEvent;
 import no.nav.fo.veilarbdialog.domain.DialogData;
-import no.nav.sbl.jdbc.Database;
-import no.nav.sbl.sql.SqlUtils;
+import no.nav.fo.veilarbdialog.auth.AuthService;
+import org.springframework.dao.EmptyResultDataAccessException;
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Component;
 import org.springframework.transaction.annotation.Transactional;
 
-import javax.inject.Inject;
-import java.sql.ResultSet;
-import java.sql.SQLException;
 import java.util.Date;
-
-import static no.nav.sbl.sql.DbConstants.CURRENT_TIMESTAMP;
+import java.util.Optional;
 
 @Component
+@RequiredArgsConstructor
 public class DataVarehusDAO {
 
-    static final String EVENT_ID = "event_id";
-    static final String DIALOGID = "dialogid";
-    static final String EVENT = "event";
-    static final String TIDSPUNKT = "tidspunkt";
-    static final String AKTOR_ID = "aktor_id";
-    static final String AKTIVITET_ID = "aktivitet_id";
-    static final String LAGT_INN_AV = "lagt_inn_av";
-    static final String EVENT_TABELL = "EVENT";
-
     private final JdbcTemplate jdbc;
-    private final Database database;
-
-    @Inject
-    public DataVarehusDAO(JdbcTemplate jdbcTemplate, Database database) {
-        this.jdbc = jdbcTemplate;
-        this.database = database;
-    }
+    private final AuthService auth;
 
     public void insertEvent(DialogData dialogData, DatavarehusEvent datavarehusEvent) {
-        Long nextSeq = SqlUtils.select(jdbc, "dual", DataVarehusDAO::hentLongFraForsteKollone)
-                .column("EVENT_ID_SEQ.NEXTVAL")
-                .execute();
-
-        SqlUtils.insert(jdbc, EVENT_TABELL)
-                .value(EVENT_ID, nextSeq)
-                .value(DIALOGID, dialogData.getId())
-                .value(EVENT, datavarehusEvent.toString())
-                .value(TIDSPUNKT, CURRENT_TIMESTAMP)
-                .value(AKTOR_ID, dialogData.getAktorId())
-                .value(AKTIVITET_ID, dialogData.getAktivitetId())
-                .value(LAGT_INN_AV, getLagtInnAv())
-                .execute();
+        long nextId = Optional
+                .ofNullable(jdbc.queryForObject("select EVENT_ID_SEQ.NEXTVAL from dual", Long.class))
+                .orElseThrow(IllegalStateException::new);
+        jdbc.update("insert into EVENT (EVENT_ID, DIALOGID, EVENT, TIDSPUNKT, AKTOR_ID, AKTIVITET_ID, LAGT_INN_AV) values (?, ?, ?, CURRENT_TIMESTAMP, ?, ?, ?)",
+                nextId,
+                dialogData.getId(),
+                datavarehusEvent.toString(),
+                dialogData.getAktorId(),
+                dialogData.getAktivitetId(),
+                auth.getIdent().orElse("SYSTEM"));
     }
 
     @Transactional(readOnly = true)
     public Date hentSisteEndringSomIkkeErDine(String aktorId, String bruker) {
-        return database.query(
```

Comment:
Kunne kanskje hatt en test på det her, er hvertfall ikke opplagt for meg at før og nå koden er lik

Reply:
Og datohåndtering, tar ikke for gitt at det blir riktig

Reply:
Nei, koden kan ikke være lik fordi `Database` er borte og vi må gjenskape bruken av `JdbcTemplate` i egen applikasjon nå.

Datohåndteringen mente @hakonph at vi burde unngått i denne PR'en, så det er jammen ikke lett å vite hva som skal til.

Det er en test som håndterer dette, i `DialogRessursTest` (siden jeg synes det er greit å teste via REST-APIet på microservices, for da gjennomgår du både controller, service og what have you). Holder riktignok på å forbedre den.

Reply:
Ok, da skal jeg ta en titt på den testen. Kan hende jeg kommenterer feil mtp tester her også fordi jeg ser igjennom filene i den rekkefølgen de står og kommenterer det som faller meg inn med en gang uten å scrolle for å finne andre filer